packages/cdktf/lib/testing/matchers.ts

```diff
@@ -0,0 +1,179 @@
+import * as fs from "fs";
+import * as path from "path";
+import { execSync } from "child_process";
+
+const terraformBinaryName = process.env.TERRAFORM_BINARY_NAME || "terraform";
+
+export interface TerraformConstructor {
+  new (...args: any[]): any;
+  terraformResourceType: string;
+}
+export type SynthesizedStack = {
+  resource: Record<string, any>;
+  data: Record<string, any>;
+};
+export type MatcherReturn = { message: () => string; pass: boolean };
+
+function assertElementWithProperties(
+  type: keyof SynthesizedStack,
+  received: string,
+  itemType: TerraformConstructor,
+  properties: Record<string, any> = {}
+) {
+  const items = Object.entries(
+    (JSON.parse(received) as SynthesizedStack)[type] || {}
+  );
+
+  const pass = items.some(([type, values]) => {
+    if (type !== itemType.terraformResourceType) {
+      return false;
+    }
+
+    return Object.values(values).some((item: any) => {
+      return Object.entries(properties).every(([key, value]) => {
+        // TODO: snake case the key
+        return item[key] === value;
+      });
+    });
+  });
+
+  if (pass) {
+    return {
+      pass,
+      message: () =>
+        `Expected ${
+          itemType.terraformResourceType
+        } not to be present in synthesised stack with properties ${JSON.stringify(
+          properties
+        )}`,
+    };
+  } else {
+    return {
+      message: () =>
+        `Expected ${
+          itemType.terraformResourceType
+        } to be present in synthesised stack with properties ${JSON.stringify(
+          properties
+        )}`,
+      pass,
+    };
+  }
+}
+
+export function toHaveDataSourceWithProperties(
+  received: string,
+  resourceType: TerraformConstructor,
+  properties: Record<string, any> = {}
+): MatcherReturn {
+  return assertElementWithProperties(
+    "data",
+    received,
+    resourceType,
+    properties
+  );
+}
+
+export function toHaveResourceWithProperties(
+  received: string,
+  resourceType: TerraformConstructor,
+  properties: Record<string, any> = {}
+): MatcherReturn {
+  return assertElementWithProperties(
+    "resource",
+    received,
+    resourceType,
+    properties
+  );
+}
+
+export function toBeValidTerraform(received: string): MatcherReturn {
+  try {
+    if (!fs.statSync(received).isDirectory()) {
+      throw new Error("Path is not a directory");
+    }
+  } catch (e) {
+    return {
+      message: () => `Expected subject to be a terraform directory: ${e}`,
+      pass: false,
+    };
+  }
+
+  try {
+    const manifest = JSON.parse(
+      fs.readFileSync(path.resolve(received, "manifest.json"), "utf8")
+    );
+
+    const stacks = Object.entries(manifest.stacks);
+
+    stacks.forEach(([name, stack]) => {
+      const opts = {
+        cwd: path.resolve(received, (stack as any).workingDirectory),
+        env: process.env,
+      } as any;
+      execSync(`${terraformBinaryName} init`, opts);
```

Comment:
Won't this print a lot of noise to the stdout that is shared with the test runner?

Reply:
I have no idea why, but if I add `stdio: "ignore"` it fails 🙈  It seems to only print sth on error and thought that would be no bigger problem